.github/actions/setup-tools/action.yaml

```
@@ -0,0 +1,38 @@
+name: Setup SUI CLI
+description: 'Setup Sui CLI'
+
+inputs:
+  install-sui:
+    description: "The flag to trigger the action to install SUI's dependency"
+    required: false
+    default: 'false'
+  install-nodejs:
+    description: "The flag to trigger the action to install nodejs dependency"
+    required: false
+    default: 'false'
+
+runs:
+  using: 'composite'  
+  steps:
+
+    # This step ensures that the "LITERAL" value is given to the actions.
+    # Using `jq` sends it uninterepted to the action which is sort of "hackish" and doesn't ensure the separation of concerns.
+    - name: Get SUI Version
+      shell: bash
+      if: inputs.install-sui == 'true'
+      run: |
+        SUI_VERSION=$(jq -r '.SUI_VERSION' "version.json")
+        echo "SUI_VERSION=$SUI_VERSION" >> $GITHUB_ENV
+
+    - name: Setup Sui CLI and install dependencies
+      if: inputs.install-sui == 'true'
+      uses: ./.github/actions/setup-sui
+      with:
+        sui-version: ${{ env.SUI_VERSION }}
```

```suggestion
        sui-version: $(jq -r '.SUI_VERSION' "version.json")
```
just read it directly here instead of having a separate step?